R/ci_setup.R

```diff
@@ -0,0 +1,89 @@
+##' @title  RcppDeepState-action GitHub workflow setup
+##' @param repository path to the repository root folder
+##' @param event the event that triggers the workflow to run, accepted values 
+##' are pull_request and push
+##' @param fail_ci_if_error specify if CI pipeline should fail when 
+##' RcppDeepState finds errors
+##' @param location location of the package if not in the root of the repository
+##' @param seed seed used for deterministic fuzz testing and reproduce the 
+##' analysis results
+##' @param time_limit fuzzing phase's duration in seconds
+##' @param max_inputs number of inputs generated in the fuzzing phase to analyze
+##' @param comment print the analysis results as a comment in pull request
+##' @param verbose enable the verbose logging mode of RcppDeepState
+##' @export
+ci_setup <- function(repository="./", event="pull_request", 
+                     fail_ci_if_error=FALSE, location="/", seed="-1", 
+                     time_limit="2", max_inputs="3", comment=FALSE, 
+                     verbose=TRUE){
+                    
+  workflow_path <- gsub("[/]+", "/", file.path(repository, ".github/workflows"))
+  workflow_file <- file.path(workflow_path, "RcppDeepState.yaml")
+  
+  # repository containing the RcppDeepState-action
+  action_repo <- "FabrizioSandri/RcppDeepState-action@main"
```

Comment:
maybe make into a function arg?

Reply:
Yes, this makes the RcppDeepstate `ci_setup` call more user-customizable.

However, I only added the `workflow_file_name` option because the `action_repo` variable should not be changed: the `ci_setup` function builds a workflow for [FabrizioSandri/RcppDeepState-action](https://github.com/FabrizioSandri/RcppDeepState-action), thus specifying a different repository makes no sense. 

Added this changes in feef454